### PR TITLE
fix problems with channels_first

### PR DESCRIPTION
**Pull request by AlexeyKozhevin**
Fix problems with kwargs in upsample.

### CODE REVIEW

File: dataset/models/tf/base.py [hunk 1]
```diff
@@ -617,6 +617,20 @@ def num_channels(self, tensor, **kwargs):
         channels_axis = self.channels_axis(tensor, **kwargs)
         return shape[channels_axis] if shape else None
 
+    @staticmethod
```
Comment (roman-kh) [inline dataset/models/tf/base.py]: why static? classmethod fits here, doesn't it?

File: dataset/models/tf/base.py [hunk 2]
```diff
@@ -1166,7 +1180,8 @@ def _add_output_predictions(self, inputs, scope, attr_prefix, **kwargs):
 
     def _add_output_proba(self, inputs, scope, attr_prefix, **kwargs):
         _ = scope, kwargs
-        proba = tf.nn.softmax(inputs, name='predicted_proba')
+        axis = -1 if kwargs['data_format'] == 'channels_last' else 1
```
Comment (roman-kh) [inline dataset/models/tf/base.py]: cls.channels_axis(tensor, **kwargs)

File: dataset/models/tf/linknet.py [hunk 3]
```diff
@@ -124,7 +124,7 @@ def decoder_block(cls, inputs, filters, name, **kwargs):
         """
         upsample_args = cls.pop('upsample', kwargs)
 
-        num_filters = inputs.get_shape()[-1].value // 4
+        num_filters = cls.get_num_channels(inputs, kwargs.get('data_format')) // 4
```
Comment (roman-kh) [inline dataset/models/tf/linknet.py]: why `get_`? there is a `num_channels` method.

Comment (AlexeyKozhevin) [reply dataset/models/tf/linknet.py]: Static 'num_channels' exists in 'data_format' branch but in 'nn' num_channels isn't static method.

File: dataset/models/tf/base.py (hunk 2 above)
Comment (AlexeyKozhevin) [reply dataset/models/tf/base.py]: Yes, I forgot about it.

File: dataset/models/tf/base.py (hunk 1 above)
Comment (AlexeyKozhevin) [reply dataset/models/tf/base.py]: No, that method is used in linknet decoder_block so it must be static. In that branch I don't rename non-static methods into 'get_...'

Comment (AlexeyKozhevin) [reply dataset/models/tf/base.py]: You're right.

**Pull request closed without merging**